File: tools/token_bench/programs/06_word_frequency/word_freq.js
const text = "the quick brown fox jumps over the lazy dog the fox the dog and the cat sat on the mat the quick brown fox";

const words = text.split(" ");
console.log(`Total: ${words.length}`);
console.log(`Unique: ${new Set(words).size}`);

const counts = {};
for (const w of words) counts[w] = (counts[w] || 0) + 1;
for (const [word, count] of Object.entries(counts)) {
    console.log(`${word}: ${count}`);
}

const longest = words.reduce((a, w) => w.length > a.length ? w : a);
console.log(`Longest: ${longest} (${longest.length} chars)`);

const avg = words.reduce((a, w) => a + w.length, 0) / words.length;
console.log(`Avg length: ${avg}`);

File: tools/token_bench/programs/06_word_frequency/word_freq.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static void Main() {
        string text = "the quick brown fox jumps over the lazy dog the fox the dog and the cat sat on the mat the quick brown fox";

        var words = text.Split(' ');
        Console.WriteLine($"Total: {words.Length}");
        Console.WriteLine($"Unique: {words.Distinct().Count()}");

        var counts = new Dictionary<string, int>();
        foreach (var w in words)
            counts[w] = counts.GetValueOrDefault(w) + 1;
        foreach (var (word, count) in counts)
            Console.WriteLine($"{word}: {count}");

        var longest = words.OrderByDescending(w => w.Length).First();
        Console.WriteLine($"Longest: {longest} ({longest.Length} chars)");

        double avg = words.Average(w => w.Length);
        Console.WriteLine($"Avg length: {avg}");
    }
}
